utils/external_resource.go

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2023 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package utils
+
+import (
+	"context"
+	"github.com/kiegroup/container-builder/util/log"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kube-openapi/pkg/validation/errors"
+	"strings"
+)
+
+const (
+	SWFExternalResourcesCMName = "workflow-external-resources-config"
+	Types                      = "types"
+	CamelRouteType             = "CamelRoute"
+	OpenAPIType                = "OpenAPI"
+	AsyncAPIType               = "AsyncAPI"
+)
+
+func getKogitoDefaultBuilderTypes() []string {
+	return []string{CamelRouteType, OpenAPIType, AsyncAPIType}
+}
+
+/*
+The workflow-external-resources-config is structured in this way
+
+[types]:[CamelRoute,OpenAPI,AsyncAPI,CustomType] this key contains the handled types
+
+
+the other keys depending on the customer and the content must be in the following format
+type!content, the first ! is the separator between type and content.
+The first ! is searched and the chars before are considered the type and the chars after the ! the content to handle
+
+i.e.
+"CamelRoute!from("ftp:myserver/folder")
+  .to("activemq:queue:cheese");"
+
+"OpenAPI!servers:
+  - url: https://development.example.com/v1
+    description: Development server"
+
+"AsyncAPI!asyncapi: 2.0.0
+info:
+  title: Account Service
+  version: '1.0.0'
+  description: |
+    Manages user accounts in the system.
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0"
+
+*/
+
+type ExternalResource struct {
+	ResourceType string
```

Comment:
This should be an enum following your doc here:

> types]:[CamelRoute,OpenAPI,AsyncAPI,CustomType] this key contains the handled types

Reply:
This is a "container" of a single External resource, used to add/update in the configmap
`
ResourceType-> CamelRoute
ContentText->from("ftp:myserver/folder").to("activemq:queue:cheese");
Key-> workflow-01`